stripe-servant/src/Stripe/Resources.hs

```diff
@@ -158,6 +167,67 @@ data PriceCreateRecurring
   , prcIntervalCount :: Maybe Int
   } deriving (Show, Eq)
 
+newtype InvoiceId = InvoiceId {unInvoiceId :: T.Text}
+  deriving (Show, Eq, ToJSON, FromJSON, ToHttpApiData)
+
+newtype InvoiceSettings = InvoiceSettings
+  { isDefaultPaymentMethod :: Maybe T.Text
+  }
+  deriving (Show, Eq)
+
+data PaymentIntent = PaymentIntent
+  { piPaymentMethod :: Maybe T.Text,
+    piStatus :: Maybe T.Text
+  }
+  deriving (Show, Eq)
+
+data Invoice = Invoice
+  { iId :: InvoiceId,
+    iCollectionMethod :: T.Text,
+    iCustomer :: CustomerId,
+    iHostedInvoiceURL :: Maybe T.Text,
+    iCreated :: Maybe Int,
+    iAmountDue :: Maybe Int,
+    iAmountPaid :: Maybe Int,
+    iStatus :: T.Text,
+    iPaymentIntent :: Maybe PaymentIntent
+  }
+  deriving (Show, Eq)
+
+newtype PaymentMethodId = PaymentMethodId {unPaymentMethodId :: T.Text}
+  deriving (Show, Eq, ToJSON, FromJSON, ToHttpApiData)
+
+data BillingDetails = BillingDetails
+  { blEmail :: Maybe T.Text,
+    blName :: Maybe T.Text,
+    blPhone :: Maybe T.Text
+  }
+  deriving (Show, Eq)
+
+data Card = Card
+  { cExpMonth :: Int
+  , cExpYear :: Int
+  , cLast4 :: String
+  }
+  deriving (Show, Eq)
+
+data PaymentMethod = PaymentMethod
+  { pmId :: PaymentMethodId,
+    pmType :: T.Text,
+    pmBillingDetails :: BillingDetails,
+    pmCard :: Card
+  }
+  deriving (Show, Eq)
+
+data PaymentMethodCreate = PaymentMethodCreate
```

Comment:
This is focused on creating cards and Stripe recommends using Stripe.js for this setup to avoid needing to be PCI compliant. I'd suggest either removing it for now or extracting the card fields to a `CardCreate` data structure.